### PR TITLE
feat: adding initial transaction transfer metadata

### PR DESCRIPTION
**Pull request by geekbrother**
This PR adds initial transaction transfer metadata schema.

### CODE REVIEW

File: crates/yttrium/src/chain_abstraction/api/route.rs [hunk 1]
```diff
@@ -23,9 +23,21 @@ pub struct RouteRequest {
...
 pub struct Metadata {
     pub funding_from: Vec<FundingMetadata>,
+    pub initial_transaction: InitialTransactionMetadata,
     pub check_in: u64,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi_macros::Record))]
+#[serde(rename_all = "camelCase")]
+pub struct InitialTransactionMetadata {
```
Comment (chris13524) [inline crates/yttrium/src/chain_abstraction/api/route.rs:33]: Should this be some sort of enum saying ERC-20 vs swap or something? I suppose we could make a V2 route endpoint though

Comment (jakubuid) [reply crates/yttrium/src/chain_abstraction/api/route.rs:33]: Yeah let's add swap and be ready for other types in the future

Comment (jakubuid) [reply crates/yttrium/src/chain_abstraction/api/route.rs:33]: @geekbrother wdyt?

Comment (llbartekll) [reply crates/yttrium/src/chain_abstraction/api/route.rs:33]: I think we can always change the interface on the yttrium level. And not to think that future forward atm

File: crates/yttrium/src/chain_abstraction/api/route.rs [hunk 2]
```diff
@@ -23,9 +23,21 @@ pub struct RouteRequest {
...
+    pub transfer_to: Address,
+    pub amount: U256,
+    pub token_contract: Address,
+    pub symbol: String,
+    pub decimals: u8,
+}
```
[the remark this answers is not in the document]
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/api/route.rs:39]: Yeah @geekbrother we should add `impl From<InitialTransactionMetadata> for Amount`

File: crates/yttrium/src/chain_abstraction/api/route.rs (hunk 1 above)
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/api/route.rs:33]: This interface also affects the Blockchain API response. But I agree we could change that easily too with new versions

Comment (jakubuid) [reply crates/yttrium/src/chain_abstraction/api/route.rs:33]: So let's keep as is then

File: crates/yttrium/src/chain_abstraction/api/route.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (geekbrother) [reply crates/yttrium/src/chain_abstraction/api/route.rs:40]: `to_amount` implementation was added.

**Review by llbartekll — APPROVED** [review]

File: crates/yttrium/src/chain_abstraction/api/route.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (llbartekll) [reply crates/yttrium/src/chain_abstraction/api/route.rs:40]: @chris13524 how to use to_amount?

**Comment by chris13524** [comment]
This fixed RES-150

**Comment by chris13524** [comment]
Fixes RES-150

File: crates/yttrium/src/chain_abstraction/api/route.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/api/route.rs:40]: We need to add the necessary uniffi export macro for it

**Pull request merged**